pkgs/by-name/li/libnick/package.nix

```
@@ -0,0 +1,51 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  boost,
+  curl,
+  glib,
+  libsecret,
+  maddy-markdown,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "libnick";
+  version = "2024.11.1";
+
+  src = fetchFromGitHub {
+    owner = "NickvisionApps";
+    repo = "libnick";
+    rev = "${finalAttrs.version}";
+    hash = "sha256-e64f8zG+/kV4BmG76UZAhDJL6IeDMbdXwyHgjYiSJtU=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
```

```suggestion
  nativeBuildInputs = [
    cmake
    ninja
    pkg-config
  ]
  ++ lib.optionals stdenv.hostPlatform.isUnix [
    pkg-config
    validatePkgConfig
  ];
```

We can use ninja here to speed up builds a bit, and use `validatePkgConfig` as a quick smoke test for the files being generated